src/core/preload.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const setImmediate = require('async/setImmediate')
+const each = require('async/each')
+const toUri = require('multiaddr-to-uri')
+const debug = require('debug')
+const CID = require('cids')
+const preload = require('./runtime/preload-nodejs')
+
+const log = debug('jsipfs:preload')
+log.error = debug('jsipfs:preload:error')
+
+// Tools like IPFS Companion redirect requests to IPFS gateways to your local
+// gateway. This is a hint to those tools that they shouldn't redirect these
+// requests as they will effectively disable the preloading.
+const redirectOptOutHint = 'x-ipfs-preload'
```

Comment:
Something to note is that browser Companion expects [`x-ipfs-companion-no-redirect`](https://github.com/ipfs-shipyard/ipfs-companion/blob/a6583ae8065640b367232d819c509af75e3002b1/add-on/src/lib/ipfs-request.js#L6).
If it does not see it in URL, it will redirect to local gateway, which we don't want (local node on a laptop may go offline).

We can add another URL hint to Companion, but is it necessary?
To limit the number of moving pieces before Dweb Summit I would just use `x-ipfs-companion-no-redirect` for now.